programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1304,7 +1304,8 @@ static void on_event(const base_event *ev) {
                     if (svc->Addresses != NULL) {
                         for (int i = 0; svc->Addresses[i]; i++) {
                             tunnel_address *addr = svc->Addresses[i];
-                            if (addr->IsHost && model_map_get(&hostnamesToAdd, addr->HostName) == NULL) {
+                            bool has_dial = ziti_service_has_permission(svc_ev->added_services[svc_idx], ziti_session_type_Dial);
+                            if (addr->IsHost && model_map_get(&hostnamesToAdd, addr->HostName) == NULL && svc_ev->added_services && has_dial) {
```

Comment:
You can remove `&& svc_ev->added_services` from the test, because that's already been tested (line 1293)